### DISCUSSION TITLE
Unable to access mvikiotlin classes in iOS app

### DISCUSSION THREAD

**Discussion by jadhavd18** [arkivanov/MVIKotlin#88] [Q&A]
After publishing library via SPM for iOS , my KMM module classes I am able to access and use correctly while its transitive dependancy i.e mvikotlin classes are not available.

To confirm in .framework file header file I didn't find MviKotlin classes names . While in your sample proj I can see the class names present in its framework header .

Is there anything I am missing to mention in gradle file so mvikotlin is not being added in my frameworks header file .
<img width="1168" alt="Screenshot 2023-03-04 at 12 16 51 PM" src="https://user-images.githubusercontent.com/114410390/222880764-d4b0a59b-4aa2-40d8-a577-39eb0f020e82.png">
<img width="1160" alt="Screenshot 2023-03-04 at 12 17 00 PM" src="https://user-images.githubusercontent.com/114410390/222880766-6778061a-3e45-41f9-95fe-815080e35fd9.png">
<img width="1165" alt="Screenshot 2023-03-04 at 12 17 11 PM" src="https://user-images.githubusercontent.com/114410390/222880775-39ff7dde-b9d1-48ae-8bdf-6dbcae292b71.png">

**Comment by arkivanov** [comment]
Can you elaborate what Libra are you publishing via SPM? And where exactly MVIKotlin classes are not available? MVIKotlin is published to maven central. It should be added to your shared module a usual via dependencies, and then you assemble your framework.

**Reply by jadhavd18** [reply]
Answers to your Questions
-Shared module framework I am trying to publish for iOS via SPM.
-MviKotlin classes are not available in iOS project where shared framework is integrated. 

On the same note I need to know -  In your sample app when dependancy added locally and built the iOS framework , MviKotlin classes are accessible in iOS app similarly if mvikotlin dependancy added via maven (remotely) and built the framework, this resulted the same issue i.e mvikotlin classes are gone missing. 

Please refer screenshots for both

<img width="1728" alt="Screenshot 2023-03-06 at 11 52 02 AM" src="https://user-images.githubusercontent.com/114410390/223035299-4e58a087-fa80-4abe-b50e-cbb05fc34752.png">

<img width="1728" alt="Screenshot 2023-03-06 at 11 53 48 AM" src="https://user-images.githubusercontent.com/114410390/223035344-714f0262-6c6a-4077-84a2-c0c53fd7d243.png">

**Comment by arkivanov** [comment]
Thanks for the clarification. I have no idea how to integrate a shared Gradle module into an iOS app via SPM. I think MVIKotlin is no different from any other KMP library here. You may need to build and publish MVIKotlin from sources manually, but that's just my guess.

In samples, the shared module is integrated directly as one self-contained framework via [embedAndSignAppleFrameworkForXcode](https://kotlinlang.org/docs/multiplatform-mobile-integrate-in-existing-app.html#connect-the-framework-to-your-ios-project). Another option is to [use Cocoapods](https://kotlinlang.org/docs/native-cocoapods.html). These two options are officially supported by KMP and documented.
